sanity/schemas/documents/post.ts

```diff
@@ -17,95 +13,144 @@ import authorType from "./author";
-    }),
-    defineField({
-      name: "author",
-      title: "Author",
-      type: "reference",
-      to: [{ type: authorType.name }],
-    }),
-  ],
-  preview: {
-    select: {
-      title: "title",
-      author: "author.name",
-      date: "date",
-      media: "coverImage",
-    },
-    prepare({ title, media, author, date }) {
-      const subtitles = [
-        author && `by ${author}`,
-        date && `on ${format(parseISO(date), "LLL d, yyyy")}`,
-      ].filter(Boolean);
-
-      return { title, media, subtitle: subtitles.join(" ") };
-    },
-  },
+	type: "document",
+	name: "post",
+	title: "Template - Blog Post",
+	description: "",
+	fields: [
+		defineField({
+			name: "title",
+			type: "string",
+			title: "Title",
+			hidden: false,
+			validation: (Rule) => Rule.required(),
+		}),
+		defineField({
+			name: "slug",
+			type: "slug",
+			title: "Slug",
+			hidden: false,
+			validation: (Rule) => Rule.required(),
+			options: { source: "title" },
+		}),
+		defineField({
+			name: "mainImage",
+			type: "image",
+			title: "Main Image",
+			hidden: false,
+		}),
+		defineField({
+			name: "author",
+			type: "reference",
+			title: "Author",
+			hidden: false,
+			to: [{ type: "author" }],
+		}),
+		defineField({
+			name: "categories",
+			type: "array",
+			of: [{ type: "string" }],
+			title: "Categories",
+			hidden: false,
+			options: { layout: "tags" },
+		}),
+		defineField({
+			name: "metadataDescription",
+			type: "text",
+			title: "Metadata Description",
+			hidden: false,
+			description:
+				"used when shared via social media, and by some search engines",
+		}),
+		defineField({
+			name: "articleText",
+			type: "array",
+			of: [
+				{
+					type: "block",
+					styles: [
+						{ title: "Heading 1", value: "h1" },
+						{ title: "Heading 2", value: "h2" },
+					],
+					lists: [
+						{ title: "Bullet", value: "bullet" },
+						{ title: "Numbered", value: "number" },
+					],
+					marks: {
+						decorators: [
+							{ title: "Strong", value: "strong" },
+							{ title: "Emphasis", value: "em" },
+							{ title: "Code", value: "code" },
+							{ title: "Underline", value: "underline" },
+						],
+						annotations: [
+							{
+								type: "object",
+								name: "link",
+								title: "url",
+								fields: [
+									defineField({
+										type: "string",
+										name: "href",
+										title: "URL",
+										validation: (Rule) => Rule.required(),
+									}),
+									defineField({
+										type: "string",
+										name: "target",
+										title: "Target",
+										options: {
+											list: [
+												{ value: "_blank", title: "Blank" },
+												{ value: "_parent", title: "Parent" },
+											],
+										},
+									}),
+								],
+							},
+						],
+					},
+				},
+				{
+					type: "reference",
+					title: "Reference",
+					to: [{ type: "ctaSchema" }, { type: "card" }, { type: "youtube" }],
```

Comment:
for most of these, it makes more sense to 'embed them' inline instead of using references. the only reason we used references for these before is because of contentful limitations

but in sanity, you can nest fields - which means you don't need to use references if you don't want to

anyhow I'm not expecting you to change this right now, but you should definitely come back to it at some point